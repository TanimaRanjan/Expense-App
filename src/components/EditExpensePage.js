import React from 'react'
import { connect} from 'react-redux'
import ExpenseForm from './ExpenseForm'
import { startEditExpense , startRemoveExpense} from '../actions/expenses';

// refactor editexpese page to be a class based componenet
// setup match 

export class EditExpensePage extends React.Component {
    onSubmit = ( expense ) => {
        this.props.startEditExpense(this.props.expense.id, expense)
        this.props.history.push('/dashboard')
    }

    onRemove =  (  ) => {
        this.props.startRemoveExpense({id: this.props.expense.id})
        this.props.history.push('/dashboard')
    }
    render() {
        return (
            <div>
                <div className='page-header'>
                    <div className='content-container'>
                        <h1 className='page-header__title'>Edit Expense</h1>
                    </div>
                </div>
                <div className='content-container'>
                    <ExpenseForm 
                        expense={this.props.expense}
                        onSubmit={this.onSubmit}
                    />
                    <button className='button-layout__remove' onClick={this.onRemove}>Remove Expense</button>
                </div>
        </div>
        )
    }
}


const mapDistpatchToProps = (dispatch, props)  => ({
    startEditExpense: (id, expense) => dispatch(startEditExpense(id, expense)),
        startRemoveExpense: (id) => dispatch(startRemoveExpense(id))
})

const mapStateToProp = (state, props) => ({
        expense: state.expenses.find((expense) => {
            return expense.id === props.match.params.id
        })
})

export default connect(mapStateToProp, mapDistpatchToProps)(EditExpensePage);