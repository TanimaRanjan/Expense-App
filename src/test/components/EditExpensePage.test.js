import React from 'react'
import { shallow } from 'enzyme'
import EditExpensePage from '../../components/EditExpensePage'
import expenses from '../fixtures/expenses'


let editExpense, removeExpense, history, wrapper 

beforeEach(() => {
    editExpense = jest.fn();
    removeExpense = jest.fn();
    history = { push : jest.fn()}
    wrapper = shallow(<EditExpensePage 
            editExpense={editExpense} 
            removeExpense={removeExpense} 
            history={history} 
            expense={expenses[2]}/>)
})
test('Should render EditExpensePage properly', () => {
    expect(wrapper).toMatchSnapshot()
})

test('Should render EditExpensePage properly', () => {
    wrapper.find('ExpenseForm').prop('onSubmit')(expenses[2])
     expect(history.push).toHaveBeenLastCalledWith('/')
})

test('Should render EditExpensePage properly', () => {
     expect(wrapper).toMatchSnapshot()
})

