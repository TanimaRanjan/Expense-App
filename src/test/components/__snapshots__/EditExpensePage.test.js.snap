// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render EditExpensePage properly 1`] = `undefined`;

exports[`Should render EditExpensePage properly 2`] = `undefined`;
